Abstract (#6)
greenelab/optimizer-manuscript: content/01.abstract.md

## Abstract {.page_break_before}

### Motivation

> Most machine learning models can be fit to data using a variety of different optimization approaches.

Reviewer: ```suggestion
Most models can be fit to data using various optimization approaches.
```

> In practice, however, studies performing classification on gene expression samples typically use a single implementation, and optimization choices are not always clear in software documentation.

Reviewer: ```suggestion
While model choice is frequently reported in machine-learning-based research, optimizers are not often noted.
```

We applied two different implementations of LASSO logistic regression implemented in Python's scikit-learn package, using two different optimization approaches (coordinate descent and stochastic gradient descent), to predict driver mutation presence or absence from gene expression across 84 pan-cancer driver genes.
Across varying levels of regularization, we compared performance and model sparsity between optimizers.

### Results

In general, we found that coordinate descent (implemented in the `liblinear` library) tended to outperform SGD for the best-performing level of regularization.

> We also found that for most driver genes, the best-performing `liblinear` model tended to be more highly regularized than the best-performing SGD model, and SGD models generally resisted overfitting as regularization strength decreased and model complexity increased.

Reviewer: ```suggestion
For most driver genes, the best-performing `liblinear` model was more highly regularized than the best-performing SGD model.
However, SGD models generally resisted overfitting as regularization strength decreased and model complexity increased.
```

> Our SGD results provide a counterexample to the conventional wisdom in classification for gene expression data, suggesting that explicit regularization is not always necessary to address high dimensionality and redundancy in the data.

Reviewer: ```suggestion
While `liblinear` results for this problem match the conventional wisdom of overfitting, SGD results contradict it.
The choice of optimizers should be clearly reported alongside model selection to allow readers and reveiwers to better understand the context in which results have been generated.
```
